Cognite.Jetfire.Cli/Deploy/Manifest/OidcCredentials.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using Cognite.Jetfire.Api.Model;
+
+namespace Cognite.Jetfire.Cli.Deploy.Manifest
+{
+    public class OidcCredentials
+    {
+        public string ClientId { get; set; }
+        public string ClientSecret { get; set; }
+        public string[] Scopes { get; set; }
+        public string TokenUrl { get; set; }
+        public string CdfProjectName { get; set; }
+    }
+
+    public class ReadWriteOidcCredentials : OidcCredentials
+    {
+        private OidcCredentials read;
+        public OidcCredentials Read
+        {
+            get
+            {
+                if (this.read != null)
+                    return this.read;
+                return this;
```

Comment:
Nit picking. This is bit weird imo. I guess it works as intended. But it is not obvious to the DTO consumer that it falls to the base class values if the Read/Write objects are not present.

Reply:
It's a workaround to facilitate both of these two schemas:
``` yaml
# Read/write the same, custom get-ers makes it so that authentication.read and .write returns this
authentication:
  tokenUrl: "https://login.microsoftonline.com/b86328db-09aa-4f0e-9a03-0136f604d20a/oauth2/v2.0/token"
  clientId: COGNITE_CLIENT_ID
  clientSecret: COGNITE_CLIENT_SECRET
  scopes:
    - https://bluefield.cognitedata.com/.default
  cdfProjectName: extractors-bluefield-testing

# Different read and write
authentication:
  read:
    tokenUrl: "https://login.microsoftonline.com/b86328db-09aa-4f0e-9a03-0136f604d20a/oauth2/v2.0/token"
    clientId: COGNITE_CLIENT_ID
    clientSecret: COGNITE_CLIENT_SECRET
    scopes:
      - https://bluefield.cognitedata.com/.default
    cdfProjectName: extractors-bluefield-testing

  write:
    tokenUrl: "https://login.microsoftonline.com/b86328db-09aa-4f0e-9a03-0136f604d20a/oauth2/v2.0/token"
    clientId: COGNITE_CLIENT_ID
    clientSecret: COGNITE_CLIENT_SECRET
    scopes:
      - https://bluefield.cognitedata.com/.default
    cdfProjectName: another-project
```
inspired by how the current schema does api-keys. If this were Python, I would be using [union types](https://github.com/cognitedata/python-extractor-utils/blob/153e1bd459e7709cc947d031ea9daf212ac55c17/cognite/extractorutils/configtools.py#L324), but I did not find anything similar for YamlDotNet.